R/dash_2.R

```diff
@@ -0,0 +1,266 @@
+# Dash 2 Syntax Functions (adapted from https://github.com/daattali/dash2)
```

Comment:
We should certainly reference @daattali's work here, but I don't think we should call this part Dash 2 as though it's a separate thing from the rest of Dash... regardless of whether we call this version 1.0 or go straight to 2.0 to match Python.

Reply:
Fixed in 0446581. I'm of the opinion that this be version 1.0, followed by 2.0 once we fully integrate the rest of Dean's changes.